spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/ConvertingMessageListener.java

```diff
@@ -0,0 +1,135 @@
+import org.apache.kafka.common.header.Header;
+
+import org.springframework.kafka.listener.AcknowledgingConsumerAwareMessageListener;
+import org.springframework.kafka.listener.AcknowledgingMessageListener;
+import org.springframework.kafka.listener.ConsumerAwareMessageListener;
+import org.springframework.kafka.listener.MessageListener;
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.converter.MessageConversionException;
+import org.springframework.messaging.converter.MessageConverter;
+import org.springframework.messaging.converter.SimpleMessageConverter;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link AcknowledgingConsumerAwareMessageListener} adapter that implements
+ * converting received {@link ConsumerRecord} using specified {@link MessageConverter}
+ * and then passes result to specified {@link MessageListener}.
+ *
+ * @param <T> the key type.
+ * @param <U> the value type.
+ * @param <V> the desired value type after conversion.
+ *
+ * @author Adrian Chlebosz
+ * @since 3.0
+ * @see AcknowledgingConsumerAwareMessageListener
+ */
+public class ConvertingMessageListener<T, U, V> implements AcknowledgingConsumerAwareMessageListener<T, U> {
+
+	private final MessageListener<T, V> delegate;
+	private final MessageConverter messageConverter;
+	private final Class<V> desiredValueType;
+
+	/**
+	 * Construct an instance with the provided {@link MessageListener} and {@link Class}
+	 * as a desired type of {@link ConsumerRecord}'s value after conversion. Default value of
+	 * {@link MessageConverter} is used, which is {@link SimpleMessageConverter}.
+	 *
+	 * @param delegate the {@link MessageListener} to use when passing converted {@link ConsumerRecord} further.
+	 * @param desiredValueType the {@link Class} setting desired type of {@link ConsumerRecord}'s value.
+	 */
+	public ConvertingMessageListener(MessageListener<T, V> delegate, Class<V> desiredValueType) {
+		Assert.notNull(delegate, "Delegate message listener cannot be null");
+		Assert.notNull(desiredValueType, "Desired value type cannot be null");
+
+		this.delegate = delegate;
+		this.desiredValueType = desiredValueType;
+
+		this.messageConverter = new SimpleMessageConverter();
+	}
+
+	/**
+	 * Construct an instance with the provided {@link MessageListener}, {@link MessageConverter} and {@link Class}
+	 * as a desired type of {@link ConsumerRecord}'s value after conversion.
+	 *
+	 * @param delegate the {@link MessageListener} to use when passing converted {@link ConsumerRecord} further.
+	 * @param messageConverter the {@link MessageConverter} to use for conversion.
+	 * @param desiredValueType the {@link Class} setting desired type of {@link ConsumerRecord}'s value.
+	 */
+	public ConvertingMessageListener(MessageListener<T, V> delegate, MessageConverter messageConverter, Class<V> desiredValueType) {
```

Comment:
If `messageConverter` is optional and we have a default one, it is better to extract it into a setter instead.

Reply:
Changed in eeb9961c

Reply:
Please, stop making comments like this.
You just push commit to the PR and we are notified. 
We also will see your changes and how they are related to the review we have left before.
Otherwise it is spamming my mail box: first your new commit and this "duplication" comments in their individual emails 😄